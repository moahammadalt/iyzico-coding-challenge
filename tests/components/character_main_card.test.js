import React, { Component } from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import CharacterMainCard from '../../src/components/character_main_card.js';


// Snapshots
describe('<CharacterMainCard /> Snapshot test for CharacterMainCard component', () => {

  it('Should match the CharacterMainCard snapshot when character data is empty', () => {

    const wrapper = shallow(<CharacterMainCard character={{}} />);

    const tree = renderer.create(wrapper).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('Should match the CharacterMainCard snapshot when character data is not empty', () => {

    const wrapper = shallow(<CharacterMainCard character={{name: 'test', image: 'test'}} />);

    const tree = renderer.create(wrapper).toJSON();

    expect(tree).toMatchSnapshot();
  });

});