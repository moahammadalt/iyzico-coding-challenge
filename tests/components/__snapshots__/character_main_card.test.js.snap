// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CharacterMainCard /> Snapshot test for CharacterMainCard component Should match the CharacterMainCard snapshot when character data is empty 1`] = `
<div
  className="character-main-card"
>
  <img
    alt="sorry, not available now"
    className="character-main-img-top rounded"
  />
  <div
    className="character-name-bottom"
  >
    <h3
      className="character-name"
    />
  </div>
</div>
`;

exports[`<CharacterMainCard /> Snapshot test for CharacterMainCard component Should match the CharacterMainCard snapshot when character data is not empty 1`] = `
<div
  className="character-main-card"
>
  <img
    alt="sorry, not available now"
    className="character-main-img-top rounded"
    src="test"
  />
  <div
    className="character-name-bottom"
  >
    <h3
      className="character-name"
    >
      test
    </h3>
  </div>
</div>
`;
